addons/account/models/account_move_line.py

```diff
@@ -2214,6 +2214,7 @@ def _check_amls_exigibility_for_reconciliation(self, shadowed_aml_values=None):
                                     This is usefull if you want to preview the reconciliation before doing some changes
                                     on amls like changing a date or an account.
         """
+        self = self.filtered(lambda aml: not aml.matching_number.startswith("P") if aml.matching_number else not aml.reconcile)
```

Comment:
I have no idea why everything is green, but there is a typo here (`reconciled`) :upside_down_face: 

And I don't understand the condition either.
Checking for `full_reconcile_id` instead of `reconciled` on the line below makes more sense to me, instead of filtering here :thinking: 

On top of that, we should check that for each group of lines in `self` grouped by `matching_number`, at least one line is not `reconciled`, following LAS's previous comment.

Reply:
I think that part is not correct.
aml1 1000 1234P reconciled=True
aml2 -999 1234P reconciled=False
aml3 -1 reconciled=False
=> aml1 is filtered out, perfect

aml1 1000 1234P reconcile=True
aml2 -999 reconciled=False
aml3 -1 reconciled=False
=> aml1 should not be filtered out, it must raise!

Reply:
Both your cases work with my grouped by `matching_number` strategy :thinking:

Reply:
@william-andre I'm not sure to understand what you are talking about :thinking:

Reply:
Note: The 2 setup depicting in my comment above could be added as tests